api/cpp/include/slint_models.h

```diff
@@ -918,8 +921,10 @@ class Repeater
             }
             auto &c = data[index];
             if (model && c.ptr) {
-                (*c.ptr)->update_data(index, *model->row_data(index));
-                c.state = State::Clean;
+                if (auto data = model->row_data(index)) {
+                    (*c.ptr)->update_data(index, *data);
+                    c.state = State::Clean;
```

Comment:
I'm not an expert here, but wouldn't it make more sense to move this statement outside the `if` block so the state is set to clean even if the model returns nullopt?

Reply:
true